### PR TITLE
feat: adding romanToInt.js

### PR DESCRIPTION
**Pull request by wesley-hsieh**
Happy easy route: Just add each letter's value to the endValue, looking up value in map data structure
Prickly route: if there is a smaller letter in front of a bigger letter, must add the difference between bigger value and smaller value to end result. 

End path: while looping, check the next letter to see if evaluates to a bigger value than the current one, if so add the difference between the next character's value and the current character's value to the end result, and skip the next character.

### CODE REVIEW

File: romanToInt.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    'X': 10,
+    'L': 50,
+    'C': 100,
+    'D': 500,
+    'M': 1000,
+};
+
+function romanToInt(x){
+    let value = 0;
+    let skipNextValue = false;
```
Comment (xaleth09) [inline romanToInt.js]: What is the Value part of this `skipNextValue`?
The above is named value, are these related?

File: romanToInt.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+    for(let i = 0; i < x.length; i++){
+        //console.log(x[i]);
```
Comment (xaleth09) [inline romanToInt.js]: nit: dont forget to remove console.logs

File: romanToInt.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        if(skipNextValue){
+            skipNextValue = false;
+            continue;
+        }
+        else{
```
Comment (xaleth09) [inline romanToInt.js]: do you need this else statement?

File: romanToInt.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+            let currLetter = x[i];
+            let nextLetter = x[i+1];
+            
+            let currValue = ROMAN_NUM_TO_INT_VALUE[currLetter];
+            let nextValue = ROMAN_NUM_TO_INT_VALUE[nextLetter];
+            
+            //console.log(`currValue ${currLetter} ${currValue} nextValue ${nextLetter} ${nextValue}`);
```
Comment (xaleth09) [inline romanToInt.js]: same as above

File: romanToInt.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    
+            if(nextValue > currValue){
+                value += nextValue - currValue;
```
Comment (xaleth09) [inline romanToInt.js]: why are we subtracting here? What is this value?

File: romanToInt.js (hunk 3 above)
Comment (xaleth09) [inline romanToInt.js]: nit: no new line between } and else
```suggestion
        } else{
```

Comment (xaleth09) [reply romanToInt.js]: non-blocking: maybe we should set up prettier for your IDE? This will run and autoformat your code for you!

File: romanToInt.js [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+                //skip the next letter/value
+                skipNextValue = true;
+            }        
+            else{
```
Comment (xaleth09) [inline romanToInt.js]: do we need this else?

**Review by xaleth09 — CHANGES_REQUESTED** [review]
some nonblockers, but some other pieces that need a little revision

File: romanToInt.js (hunk 3 above)
Comment (wesley-hsieh) [reply romanToInt.js]: Done

File: romanToInt.js (hunk 6 above)
Comment (wesley-hsieh) [reply romanToInt.js]: Yes, or the value will be double counted.

**Review by xaleth09 — APPROVED** [review]
